pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -220,6 +222,23 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 		disk.BootOrder = &api.BootOrder{Order: *diskDevice.BootOrder}
 	}
 
+	if diskDevice.BlockSize != nil {
+		if blockSize := diskDevice.BlockSize.Custom; blockSize != nil {
+			disk.BlockIO = &api.BlockIO{
+				LogicalBlockSize:  blockSize.Logical,
+				PhysicalBlockSize: blockSize.Physical,
+			}
+		} else if matchFeature := diskDevice.BlockSize.MatchVolume; matchFeature != nil && (matchFeature.Enabled == nil || *matchFeature.Enabled) {
+			blockIO, err := getOptimalBlockIO(disk)
+			if err != nil {
+				return fmt.Errorf("failed to configure disk with block size detection enabled: %v", err)
+			}
+			// TODO: If this result/value changed between VM(I) spec updates,
+			// should we block that somewhere?
```

Comment:
VMI spec updates aren't allowed. you're completely fine here.

Reply:
Ok, comment removed.